tests/test_util.py

```diff
@@ -220,32 +220,71 @@ def test_when_reading_from_queue_health_update_happens(self, parser, mock_path,
 
 
 class TestGenerateOutputFilename(unittest.TestCase):
-    def test_includes_provided_regridded_subsetted_ext(self):
+    def test_includes_provided_suffixes_ext(self):
+        """Ensure the correct combinations of regridded, subsetted and
+        reformatted are included in the correct order, per the optional
+        arguments to the function."""
         url = 'https://example.com/fake-path/abc.123.nc/?query=true'
         ext = 'zarr'
 
         # Basic cases
         variables = []
-        self.assertEqual(
-            util.generate_output_filename(url, ext),
-            'abc.123.zarr'
-        )
-        self.assertEqual(
-            util.generate_output_filename(url, ext, is_subsetted=True),
-            'abc.123_subsetted.zarr'
-        )
-        self.assertEqual(
-            util.generate_output_filename(url, ext, is_regridded=True),
-            'abc.123_regridded.zarr'
-        )
-        self.assertEqual(
-            util.generate_output_filename(url, ext, is_subsetted=True, is_regridded=True),
-            'abc.123_regridded_subsetted.zarr'
-        )
-        self.assertEqual(
-            util.generate_output_filename(url, ext, variable_subset=variables, is_subsetted=True, is_regridded=True),
-            'abc.123_regridded_subsetted.zarr'
-        )
+        with self.subTest('No suffix options'):
```

Comment:
It possibly seems a bit of overkill to add all the `self.subTest` context managers, but the nice thing is that if one of the assertions now fail, it will tell you which subtest it was in.